release/nightly_tests/dataset/image_loader_microbenchmark.py

```diff
@@ -31,10 +35,11 @@ def iterate(dataset, label, batch_size, metrics, output_file=None):
     num_rows = 0
     print_at = 1000
     for batch in it:
-        # note(swang): this will be slightly off if batch_size does not divide
-        # evenly into number of images but should be okay for large enough
-        # datasets.
-        num_rows += batch_size
+        if isinstance(batch, tuple) or isinstance(batch, list):
+            batch = batch[0]
+        else:
+            batch = batch["image"]
+        num_rows += batch.shape[0]
```

Comment:
this is a bit hard to understand. can you pass a function that returns num_rows for the batch as a parameter?